tests/network/vmi_networking.go

```diff
@@ -284,21 +283,17 @@ var _ = SIGDescribe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:c
 		})
 	})
 
-	Context("VirtualMachineInstance with disabled automatic attachment of interfaces", func() {
+	Context("VirtualMachineInstance with no networks and disabled automatic attachment of interfaces", func() {
 		It("[test_id:1774]should not configure any external interfaces", func() {
-			By("checking loopback is the only guest interface")
-			autoAttach := false
-			detachedVMI := libvmifact.NewAlpine()
-			// Remove the masquerade interface to use the default bridge one
-			detachedVMI.Spec.Domain.Devices.Interfaces = nil
-			detachedVMI.Spec.Networks = nil
-			detachedVMI.Spec.Domain.Devices.AutoattachPodInterface = &autoAttach
+			By("creating a VMI with no interfaces and networks")
+			detachedVMI := libvmifact.NewAlpine(libvmi.WithAutoAttachPodInterface(false))
 
 			var err error
 			detachedVMI, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), detachedVMI, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			libwait.WaitUntilVMIReady(detachedVMI, console.LoginToAlpine)
 
+			By("checking that loopback is the only guest interface")
 			err = console.SafeExpectBatch(detachedVMI, []expect.Batcher{
```

Comment:
I would expect to first check that no interfaces are observed on the VMI status.
Only then, to go into the guest and confirm that the status is indeed correct (i.e. checking that the status is reporting correctly that there are no vNICs).